lisa/node.py

```diff
@@ -1076,6 +1079,15 @@ def list(self) -> Iterable[Node]:
         for node in self._list:
             yield node
 
+    def list_features_by_type(
+        self, type_name: Type[T_SETTINGS]
+    ) -> Iterable[T_SETTINGS]:
+        for node in self._list:
+            if node.capability.features:
```

Comment:
1. Why this method is needed? The node.features should be enough.
1. One node should have only one feature with the same name. If it needs to present multiple information, it should be in the feature itself. So if a method get type_name as an argument, it should return only one feature.

Reply:
environment.nodes does not have features, it only has capability.features so I couldn't make use of __getitem__ in Features or get_feature_settings.

This function returns multiple type settings because it is implemented in Nodes. It gets the feature settings for all the nodes.

Reply:
I see. It looks the availability feature is for an environment, not a node. So, it's confusing. For simple, we can assume every node has the same feature, and only pick it from first node. In future, if it's too complex, we can consider supporting features on the environment level.